sqlglot/parser.py

```diff
@@ -2565,6 +2567,24 @@ def _parse_select_or_expression(self):
     def _parse_use(self):
         return self.expression(exp.Use, this=self._parse_id_var())
 
+    def _parse_transaction(self):
+        this = None
+        if (
+            self._match_text("DEFERRED")
+            or self._match_text("IMMEDIATE")
+            or self._match_text("EXCLUSIVE")
+        ):
+            this = self._prev.text
+
+        self._match_text("TRANSACTION") or self._match_text("WORK")
+
+        modes = []
+        while self._match(TokenType.VAR) or self._match(TokenType.COMMA):
+            modes.append(self._prev.text)
+
+        modes = " ".join(modes).split(" , ") if modes else None
```

Comment:
i don't think we should convert it into a string can we map the different modes and make them first class?

Reply:
I'm not sure I understood what you have in mind. So maybe create a "Mode" expression node and store a list of these with `this` set to the corresponding string?

What about `Transaction.this`? Do we want to also store that as a string literal node, or is it fine that way?

Reply:
discussed in slack